src/services/notification.service.ts

```diff
@@ -9,12 +9,16 @@ interface InsertNotificationResult{
   notification_id: number;
 }
 
-const createNotification = async (notification: Notification): Promise<number> => {
+const createNotification = async (notification: Notification): Promise<unknown> => {
   const result = await db.sql<InsertNotificationResult>('insert-notification', {
     to_user_id: notification.toUserId,
     notification_type: notification.notificationType,
   });
```

Comment:
You mentioned maybe doing an error check, which I think is a good idea.  I ran:

```
curl --header "Content-Type: application/json" \
  --request POST \
  --data '{"toUserId":"1","notificationType":"randomnotification"}' \
  http://localhost:3000/api/notifications
```
Which was successful.  I can see that on a successful request, `result` includes a `row_count`.  You might want to include a check that `row_count` > 0, since this call should always insert something.

```
curl --header "Content-Type: application/json" \
  --request POST \
  --data '{"toUserId":null,"notificationType":"randomnotification"}' \
  http://localhost:3000/api/notifications
```

This returns an error since `toUserId` can't be null.  The error looks like:

```
{
  "error":{
    "_original":{
      "toUserId":null,
      "notificationType":"randomnotification"
    },
    "details":[
      {
        "message":"\"toUserId\" must be a number",
        "path":[
          "toUserId"
        ],
        "type":"number.base",
        "context":{
          "label":"toUserId",
          "value":null,
          "key":"toUserId"
        }
      }
    ]
  }
}
```

So you could also check for the presence of `error` in the response, probably with a `try...catch` block as in the health check.

Reply:
Thanks for directing  me correctly . I was doing an error check with if else and hasOwnProperty. Try... catch is a better solution.